src/sinks/loki.rs

```diff
@@ -55,6 +57,19 @@ pub struct LokiConfig {
     tls: Option<TlsOptions>,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum OutOfOrderAction {
+    Drop,
+    RewriteTimestamp,
+}
+
+impl Default for OutOfOrderAction {
+    fn default() -> OutOfOrderAction {
+        OutOfOrderAction::RewriteTimestamp
```

Comment:
I think failure might actually be a better default as it preserves data integrity. The user can then decide if they are ok rewriting the timestamp.

Reply:
I'd tend to agree.

Reply:
Sorry, I don't think I made my preference clear. I think the default should be a new `OutOfOrderAction` which is `OutOfOrderAction::Error` that would essentially persist the current behavior of bubbling up out-of-order errors to the user. They can then decide if they want to drop or rewrite timestamps.

I think the disadvantage of:

* `Drop` as a default is that the user will lose data without realizing that is a possibility
* `RewriteTimestamp` as the default is that the event data will be mutated, again likely without the user realizing and then later being confused

Reply:
The event would still be dropped at Loki so I think it's better that we drop it here. I have a warning for this (`Received out-of-order event; dropping event.`), I prefer warning because this is something we have expected.

Reply:
If we were to generate an error - that error would be dropping the whole batch, right? Or even shutdown the whole topology on the sink error?
What do we actually want to happen when the `OutOfOrderAction::Error` mode is active? Maybe it should be `OutOfOrderAction::Ignore` (to just let `loki` error out or deal with it somehow)?